zeebe/clients/java/src/main/java/io/camunda/zeebe/client/impl/worker/JobClientImpl.java

```diff
@@ -30,25 +30,40 @@
 import io.camunda.zeebe.client.impl.command.FailJobCommandImpl;
 import io.camunda.zeebe.client.impl.command.StreamJobsCommandImpl;
 import io.camunda.zeebe.client.impl.command.ThrowErrorCommandImpl;
+import io.camunda.zeebe.client.impl.http.HttpClient;
 import io.camunda.zeebe.gateway.protocol.GatewayGrpc.GatewayStub;
 import java.util.function.Predicate;
 
 public final class JobClientImpl implements JobClient {
 
   private final GatewayStub asyncStub;
+  private final HttpClient httpClient;
   private final ZeebeClientConfiguration config;
   private final JsonMapper jsonMapper;
   private final Predicate<StatusCode> retryPredicate;
+  private final boolean useRest;
 
   public JobClientImpl(
       final GatewayStub asyncStub,
       final ZeebeClientConfiguration config,
       final JsonMapper jsonMapper,
       final Predicate<StatusCode> retryPredicate) {
+    this(asyncStub, null, config, jsonMapper, retryPredicate, false);
```

Comment:
❓ Is it safe to leave this constructor as it is? 

It might be safer to `@Deprecate` it if we're not removing it. I also see that it's only used in the `JobStreamImplTest` and we can just pass a `null` there instead of keeping the constructor.

Reply:
Yeah, good point. The constructor is not API, as far as I can see, so we can simply remove it and pass a null from the test that uses it 👍